todo_app/app.py

```diff
@@ -65,15 +104,16 @@ def index():
    @app.route('/items/add', methods = ['POST'])
    @login_required
    def add_item():
+      
       if (disable_login or current_user.role == writer_role):
          form_data = request.form
          task_title = form_data["title"]
          task_desc = form_data["description"]
          task_due_date = form_data["duedate"]
-         
+         app.logger.info("Adding a new task, details: %s | %s | %s" , task_title, task_desc, task_due_date)
          mongo_client.add_task(task_title, task_desc, task_due_date) 
       else:
-         print("Reader Role is not allowed to add a task")
+         logger.error(f"Reader Role is not allowed to add a task")
```

Comment:
This should be a warn rather than an error.

Error is only when something has really gone wrong, but here, it could just be a "reader" user trying their luck and the application is behaving perfectly fine.

An obvious example when you don't want an error even though the user action has failed is for the login process - if a user enters their password incorrectly, you don't want to log an error. That's totally on them. Having this logged as an error will just obscure real errors.

It's worth logging as a warning because if it happens a lot then that indicates you should do something.